src/backend/distributed/utils/resource_lock.c

```diff
@@ -192,6 +193,31 @@ LockShardListResources(List *shardIntervalList, LOCKMODE lockMode)
 
 
 /*
+ * LockRelationShards takes locks on all shards in a list of RelationShards
+ * to prevent concurrent DML statements on those shards.
+ */
+void
+LockRelationShardListResources(List *relationShardList, LOCKMODE lockMode)
```

Comment:
Actual function name and name in comments are different. Consider using more descriptive name. LockRelationShardListResources made me think it is locking the list, not individual shards in ShardList.